pkg/overlay/config.go

```diff
@@ -102,16 +103,18 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (
 		}
 	}()
 
-	pb.RegisterOverlayServer(server.GRPC(), &Server{
+	srv := &Server{
 		dht:   kad,
 		cache: cache,
 
 		// TODO(jt): do something else
 		logger:  zap.L(),
 		metrics: monkit.Default,
-	})
-
-	return server.Run(context.WithValue(ctx, ctxKeyOverlay, cache))
+	}
+	pb.RegisterOverlayServer(server.GRPC(), srv)
+	context.WithValue(ctx, ctxKeyOverlay, cache)
```

Comment:
this returns a new context with the values added.

https://golang.org/pkg/context/#WithValue